src/django_upgrade/fixers/http_headers.py

```diff
@@ -0,0 +1,185 @@
+"""
+Rewrite HTTP headers as headers keyword parameter:
+https://docs.djangoproject.com/en/dev/releases/4.2/#tests
+"""
+from __future__ import annotations
+
+import ast
+from functools import partial
+from typing import Iterable
+
+from tokenize_rt import Offset
+from tokenize_rt import Token
+
+from django_upgrade.ast import ast_start_offset
+from django_upgrade.data import Fixer
+from django_upgrade.data import State
+from django_upgrade.data import TokenFunc
+from django_upgrade.fixers.assert_form_error import looks_like_client_call
+from django_upgrade.tokens import find
+from django_upgrade.tokens import NAME
+from django_upgrade.tokens import OP
+from django_upgrade.tokens import parse_call_args
+
+fixer = Fixer(
+    __name__,
+    min_version=(4, 2),
+)
+
+HEADERS_KWARG = "headers"
+HTTP_PREFIX = "HTTP_"
+
+
+def merge_http_headers_kwargs(tokens: list[Token], i: int, *, node: ast.Call) -> None:
+    http_kwargs_names = []
+    contains_headers_kwarg = False
+
+    for kw in node.keywords:
+        if kw.arg and kw.arg.startswith(HTTP_PREFIX):
```

Comment:
What about `CONTENT_TYPE` and `CONTENT_LENGTH`?
Should they also be re-written to `headers` kwarg?

Reply:
Yes

Reply:
Actually maybe not? https://github.com/django/django/blob/69352d85fa8412865db9e0c7f177b333c0eac3e2/django/test/client.py#L577-L582